src/commons/api/roas.rs

```diff
@@ -280,22 +286,186 @@ impl Ord for RoaDefinition {
     }
 }
 
-impl PartialOrd for RoaDefinition {
+impl PartialOrd for RoaPayload {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         Some(self.cmp(other))
     }
 }
 
-impl AsRef<TypedPrefix> for RoaDefinition {
+impl AsRef<TypedPrefix> for RoaPayload {
     fn as_ref(&self) -> &TypedPrefix {
         &self.prefix
     }
 }
 
+//------------ RoaConfiguration --------------------------------------------
+
+/// This type defines a saved RoaConfiguration. It includes the actual ROA
+/// payload that needs be authorized in a ROA object as well as other
+/// information that is only visible to users - i.e. an optional comment
+/// field, and in future perhaps other things such as tags used for
+/// classification/monitoring/bpp analysis.
+///
+/// Note that this type is backward compatible with the RoaDefinition type
+/// used until Krill 0.10.0.
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct RoaDefinitions(Vec<RoaDefinition>);
+pub struct RoaConfiguration {
+    #[serde(flatten)]
+    payload: RoaPayload,
+    #[serde(default)] // missing is same as no comment
+    comment: Option<String>,
+}
+
+impl RoaConfiguration {
+    pub fn new(payload: RoaPayload, comment: Option<String>) -> Self {
+        RoaConfiguration { payload, comment }
+    }
+
+    pub fn unpack(self) -> (RoaPayload, Option<String>) {
```

Comment:
It's a bit late to point this out now as the `unpack` pattern is used in many places in Krill I believe, but isn't this expressable using the `From` trait?

Reply:
I suppose we could use `impl From<TypeA> for (TypeB, TypeC, ..)` like things instead. But I would make a separate issue for this.. I am using `unpack` rather than `unwrap` as these functions were called in the beginning of krill to avoid at least some confusion :)